### ISSUE TITLE
Texture Compression Colors

### ISSUE THREAD

**Issue by EclipsedButter** [Grimrukh/soulstruct-blender#75]
Ok, this one will need a bit of a preface. I am running soulstruct-blender on Mac, with wine handling `texconv.exe` and `ctypes` (zugbruecke). So huge disclaimer: this could entirely be on my end. I'll operate on the assumption that it's not for now, so please let me know if you can't reproduce this on windows.

Since wine caused issues with multiprocessing, I changed `soulstruct/containers/tpf.py.batch_get_tpf_texture_tga_data` to run sequentially instead. This meant calling `wine texconv.exe` over and over, which was very slow, since wine takes 10-15 seconds to start up *per call*. I decided to make it faster by instead bundling all of the calls to `texconv.exe` into one "&&" delimited string and passing all of that to one call of `wine cmd`, which emulates windows cmd and runs each command it was passed (delimited by &&) sequentially. This was significantly faster, but had an unintended behavior:

`Unable to find a suitable DXT compression, falling back to uncompressed (c2100_wp_a_0101_a.tga, 512x1024).`

This message popped up for 9 of the 37 textures imported. I compared the differences between my old version and the new one with the error, and you can see yourself below.

Compressed:
<img width="1440" alt="compressed" src="https://github.com/user-attachments/assets/4958093d-fcdb-47c4-900e-e5d31bbb2d73">

Uncompressed:
<img width="1440" alt="uncompressed" src="https://github.com/user-attachments/assets/11fe66e2-9ce1-440a-85b7-7acc5ba141ba">

Let's ignore that it can't find some of those textures for a moment, I'm still working on that. The compressed version has significantly darker colors, to the point of blotting out almost all detail. I checked the cache, and confirmed that they were much, much darker. Interestingly, all of the 'uncompressed' ones had significantly smaller file sizes, as low as a tenth of the compressed version.

So overall I'm not sure if this is intended, and how lossy the compression is. It would be unfortunate if the compression removed enough to make export a moot point.

**Comment by EclipsedButter** [comment]
Alright, I removed zugbruecke to run dcx decompression natively, just in case, and I can confirm that it's not an artifact of using zugbruecke.ctypes. This leaves just my particular implementation of the calls to texconv, or the fact that I am running texconv with wine.

**Comment by Grimrukh** [comment]
Hmm, this is an interesting one. My first note is that `Unable to find a suitable DXT compression, falling back to uncompressed` is a Blender warning, not a texconv one, and AFAIK it should only be logged if Blender is attempting to import a DDS file - which shouldn't be happening in this case, unless something weird is going on due to wine and/or the `&&` chaining... 

As for the texture appearance itself, it's difficult to tell in generic Blender lighting (especially in texture preview mode) and dependent on my shader reconstruction guesswork, but AFAICT these BKA (c2100) textures are indeed very dark and look as such outside of direct sunlight in-game. The converted TGA textures in Blender on my end look identical to the DDS textures opened in Visual Studio or Paint.NET. It would be great if I could see the TGA Blender textures in the Image Viewer in your second "uncompressed" case, say for `c2100_wp_a_0101_a` or `c2100_lg_a` (the darkest ones).

**Comment by EclipsedButter** [comment]
Hmmm. I guess that does make sense, since Blender is what's creating the cache, so it throwing the compression warning lines up with the changes that I see viewing the files directly, plus their file sizes (still not sure why uncompressed is smaller though...).

I've looked through both caches side-by-side, and noticed a couple things. For one thing, macOS preview can't read 6 of those 37 textures, while Blender can. I opened the files in TextEdit to compare compressed and uncompressed, and they do look significantly different. All 37 of the 'uncompressed' ones still have a DDS header, and the bytes in each file is significantly different, so I agree with you that it's probably still reading them as DDS, which means that wine texconv probably failed somewhere. Below is a 'compressed' texture viewed in Blender (left), and an 'uncompressed' texture. 
<img width="312" alt="compressedWP" src="https://github.com/user-attachments/assets/1ebd15f4-83a5-460b-ab96-0e8b95dbbc1a"><img width="312" alt="uncompressedWP" src="https://github.com/user-attachments/assets/7d2ca6da-e367-4cc0-bbb3-439d3368e3d8">


------------

Well! While writing this, I abruptly recalled a little quirk of running texconv on wine, which is that the input files have to be in the same directory as texconv. I had forgotten to do this when I reworked things, and the fact that I was getting results at all very much implied that texconv never really ran in any capacity at all. I checked, and turns out I had mis-wrote a string and was just reading the bytes from the dds files and appending that to `tga_data`. This explains why they had dds headers. Now that I've figured that out, wine cmd is being stubborn about receiving so much input at once, so I just have to get texconv working or find a native alternative. Since you said that the BKA textures are supposed to be that dark, I'll close this one.
